core/corehttp/gateway_test.go

```diff
@@ -0,0 +1,125 @@
+package corehttp
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+	b58 "github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-base58"
+	core "github.com/jbenet/go-ipfs/core"
+	coreunix "github.com/jbenet/go-ipfs/core/coreunix"
+	namesys "github.com/jbenet/go-ipfs/namesys"
+	ci "github.com/jbenet/go-ipfs/p2p/crypto"
+	repo "github.com/jbenet/go-ipfs/repo"
+	config "github.com/jbenet/go-ipfs/repo/config"
+	u "github.com/jbenet/go-ipfs/util"
+	testutil "github.com/jbenet/go-ipfs/util/testutil"
+)
+
+type mockNamesys map[string]string
+
+func (m mockNamesys) Resolve(ctx context.Context, name string) (value u.Key, err error) {
+	enc, ok := m[name]
+	if !ok {
+		return "", namesys.ErrResolveFailed
+	}
+	dec := b58.Decode(enc)
+	if len(dec) == 0 {
+		return "", fmt.Errorf("invalid b58 string for name %q: %q", name, enc)
+	}
+	return u.Key(dec), nil
+}
+
+func (m mockNamesys) CanResolve(name string) bool {
+	_, ok := m[name]
+	return ok
+}
```

Comment:
it will be written `ns.<word>(name)`

```
ns.LooksApplicable(name)
ns.LooksResolvable(name)
ns.Compatible(name)
ns.CompatibleWith(name)
ns.MayResolve(name)
ns.CanResolve(name)
ns.IsResolvable(name)
ns.Resolvable(name)
```